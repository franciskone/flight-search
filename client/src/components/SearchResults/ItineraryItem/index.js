import ItineraryItem from './ItineraryItem';

export default ItineraryItem;

export { ItineraryType } from './ItineraryItem';
