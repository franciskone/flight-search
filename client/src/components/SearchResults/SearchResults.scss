@import '~bpk-mixins';

.SearchResults {
  &__itineraries {
    padding: $bpk-spacing-sm;
  }
}
