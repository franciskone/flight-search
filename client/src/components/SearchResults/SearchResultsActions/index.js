import SearchResultsActions from './SearchResultsActions';

export default SearchResultsActions;
