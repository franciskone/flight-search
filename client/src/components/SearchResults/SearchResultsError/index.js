import SearchResultsError from './SearchResultsError';

export default SearchResultsError;
