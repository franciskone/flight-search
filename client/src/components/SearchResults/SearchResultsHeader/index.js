import SearchResultsHeader from './SearchResultsHeader';

export default SearchResultsHeader;
