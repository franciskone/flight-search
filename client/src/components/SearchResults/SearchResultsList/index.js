import SearchResultsListContainer from './SearchResultsListContainer';

export default SearchResultsListContainer;

