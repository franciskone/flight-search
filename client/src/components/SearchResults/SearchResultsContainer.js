import { connect } from 'react-redux';
import SearchResults from './SearchResults';
import {
  flightSearchHasQuerySelector,
  flightSearchIsLoadingSelector,
  flightSearchItinerariesForResultsSelector,
} from '../../features/flightSearch';

const fakeItineraries = [
  {
    id: '11235-1909211735--32302-0-13554-1909211915',
    legs: [
      {
        carrier: 'Alitalia',
        origin: {
          date: '2019-09-21T17:35:00',
          airport: 'EDI',
        },
        destination: {
          date: '2019-09-21T19:15:00',
          airport: 'LHR',
        },
        duration: 90,
        stops: 0,
      },
      {
        carrier: 'Alitalia',
        origin: {
          date: '2019-09-28T09:40:00',
          airport: 'EDI',
        },
        destination: {
          date: '2019-09-29T09:30:00',
          airport: 'LHR',
        },
        duration: 50,
        stops: 1,
      },
    ],
    price: 256,
    agent: 'omegaflightstore.com',
  },
  {
    id: '11235-1909211735--32302-0-13554-1909211917',
    flights: [
      {
        carrier: 'Alitalia',
        origin: {
          date: '2019-09-21T17:35:00',
          airport: 'EDI',
        },
        destination: {
          date: '2019-09-21T19:15:00',
          airport: 'LHR',
        },
        duration: 90,
        stops: 0,
      },
      {
        carrier: 'Alitalia',
        origin: {
          date: '2019-09-28T09:40:00',
          airport: 'EDI',
        },
        destination: {
          date: '2019-09-29T09:30:00',
          airport: 'LHR',
        },
        duration: 50,
        stops: 1,
      },
    ],
    price: 256,
    agent: 'omegaflightstore.com',
  },
  {
    id: '11235-1909211735--32302-0-13554-1909211918',
    flights: [
      {
        carrier: 'Alitalia',
        origin: {
          date: '2019-09-21T17:35:00',
          airport: 'EDI',
        },
        destination: {
          date: '2019-09-21T19:15:00',
          airport: 'LHR',
        },
        duration: 90,
        stops: 0,
      },
    ],
    price: 256,
    agent: 'omegaflightstore.com',
  },
]; // TODO Franciskone: DELETE

const mapStateToProps = state => ({
  isLoading: flightSearchIsLoadingSelector(state),
  itineraries: flightSearchItinerariesForResultsSelector(state),
  hasQuery: flightSearchHasQuerySelector(state),
});

const SearchResultsContainer = connect(mapStateToProps)(SearchResults);
export default SearchResultsContainer;
