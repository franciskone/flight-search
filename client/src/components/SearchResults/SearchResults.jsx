import React from 'react';
import PropTypes from 'prop-types';

import SearchResultsHeader from './SearchResultsHeader';
import SearchResultsActions from './SearchResultsActions';
import Spinner from '../Spinner';
import SearchResultsList from './SearchResultsList';
import SearchResultsError from './SearchResultsError';

const SearchResults = ({
  isLoading, hasQuery, errorMessage, searchItinerary, shouldSearchAgain,
}) => {
  const errorAction = shouldSearchAgain
    ? searchItinerary
    : () => window.location.reload();
  
  const errorButtonLabel = shouldSearchAgain
    ? 'Try again'
    : 'Reload the page';
  
  return ( // TODO Franciskone: add tests
    <div>
      { isLoading
        ? <Spinner label="Searching" />
        : errorMessage
          ?
            <SearchResultsError
              message={errorMessage}
              buttonLabel={errorButtonLabel}
              action={errorAction}
            />
          : hasQuery && (
          <React.Fragment>
            <SearchResultsHeader />
            <SearchResultsActions />
            <SearchResultsList />
          </React.Fragment>
        )
      }
    </div>
  );
};

SearchResults.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  hasQuery: PropTypes.bool.isRequired,
  searchItinerary: PropTypes.func.isRequired,
  shouldSearchAgain: PropTypes.bool.isRequired,
};

export default SearchResults;

