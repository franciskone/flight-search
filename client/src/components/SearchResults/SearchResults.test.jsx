import React from 'react';
import TestRenderer from 'react-test-renderer';

import SearchResults from './SearchResults';

describe('SearchResults', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(
      <SearchResults
        isLoading={false}
        hasQuery={false}
        searchItinerary={() => 'test'}
        shouldSearchAgain={false}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

