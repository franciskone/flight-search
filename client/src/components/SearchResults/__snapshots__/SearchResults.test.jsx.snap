// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should render correctly 1`] = `<div />`;
